src/components/DocumentationLayout.vue

```diff
@@ -0,0 +1,285 @@
+
+  Copyright (c) 2024 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <div class="documentation-layout">
+    <Nav
+      v-if="!isTargetIDE"
+      :diffAvailability="diffAvailability"
+      :interfaceLanguage="interfaceLanguage"
+      :objcPath="objcPath"
+      :swiftPath="swiftPath"
+      :displaySidenav="enableNavigator"
+      @toggle-sidenav="handleToggleSidenav"
+    >
+      <template #title>
+        <slot name="nav-title" />
+      </template>
+    </Nav>
+    <AdjustableSidebarWidth
+      v-bind="sidebarProps"
+      v-on="sidebarListeners"
+      class="full-width-container topic-wrapper"
+    >
+      <PortalTarget name="modal-destination" multiple />
+      <template #aside="{ scrollLockID, breakpoint }">
+        <NavigatorDataProvider
+          :interface-language="interfaceLanguage"
+          :technologyUrl="technology ? technology.url : ''"
+          :api-changes-version="selectedAPIChangesVersion"
+          ref="NavigatorDataProvider"
+        >
+          <template #default="slotProps">
+            <div class="documentation-layout-aside">
+              <QuickNavigationModal
+                v-if="enableQuickNavigation"
+                :children="slotProps.flatChildren"
+                :showQuickNavigationModal.sync="showQuickNavigationModal"
+                :technology="technology ? technology.title : ''"
+              />
+              <transition name="delay-hiding">
+                <Navigator
+                  v-show="sidenavVisibleOnMobile || breakpoint === BreakpointName.large"
+                  :flatChildren="slotProps.flatChildren"
+                  :parent-topic-identifiers="parentTopicIdentifiers"
+                  :technology="slotProps.technology || technology"
+                  :is-fetching="slotProps.isFetching"
+                  :error-fetching="slotProps.errorFetching"
+                  :api-changes="slotProps.apiChanges"
+                  :references="references"
+                  :navigator-references="slotProps.references"
+                  :scrollLockID="scrollLockID"
+                  :render-filter-on-top="breakpoint !== BreakpointName.large"
+                  @close="handleToggleSidenav(breakpoint)"
+                >
+                  <template v-if="enableQuickNavigation" #filter>
+                    <QuickNavigationButton @click.native="openQuickNavigationModal" />
+                  </template>
+                  <template #navigator-head>
+                    <slot name="nav-title" />
+                  </template>
+                </Navigator>
+              </transition>
+            </div>
+          </template>
+        </NavigatorDataProvider>
+      </template>
+      <slot name="content" />
```

Comment:
Very minor thing, but maybe this should just be the default slot instead of a named "content" slot?

This is the area where the main content of the documentation page would be rendered within this layout, right?

Reply:
It was like that before but @hqhhuang had some comments on the visibility of the slot https://github.com/apple/swift-docc-render/pull/792/#discussion_r1566403071
I really don't have a strong opinion about it. Whatever you decide it's better.